### DISCUSSION TITLE
Dropwizard 0.8.5: MySQL CommunicationsException

### DISCUSSION THREAD

**Discussion by zongbintu** [dropwizard/dropwizard#5626] [General]
version:dropwizard-db-0.8.5

jdbcurl added autoReconnect=true

```
com.mysql.jdbc.exceptions.jdbc4.CommunicationsException: The last packet successfully received from the server was 46,229,971 milliseconds ago.  The last packet sent successfully to the server was 46,229,971 milliseconds ago. is longer than the server configured value of 'wait_timeout'. You should consider either expiring and/or testing connection validity before use in your application, increasing the server configured values for client timeouts, or using the Connector/J connection property 'autoReconnect=true' to avoid this problem.
```

**Comment by zongbintu** [comment]
```
configuration.getDataSourceFactory().setUrl(mysqlUrl);
                configuration.getDataSourceFactory().setUser(mysqlUser);
                configuration.getDataSourceFactory().setPassword(mysqlPwd);
```
url:?autoReconnect=true&serverTimezone=Asia/Shanghai&autoReconnectForPools=true

**Comment by zUniQueX** [comment]
Hi @2tu. You are using a very old and outdated version of Dropwizard. I'd highly encourage you to upgrade to a supported version (2.0.x or 2.1.x) to stay secure with latest dependency updates.

The error you are describing is related to the MySQL connector and not to the Dropwizard framework itself. If this is a compatibility issue of your MySQL server and the connector in your application, upgrading your Dropwizard version may also resolve your problem.

**Comment by zongbintu** [comment]
Thanks for being in touch.I will respond as soon as possible!
